git-utils/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::{Path, PathBuf};
+
+use git2::{Branches, BranchType, Commit, Repository, Signature};
+use git2::build::RepoBuilder;
+use tempdir::TempDir;
+
+pub struct TestGit {
+    pub dir: PathBuf,
+    pub remote: Repository,
+    pub local: Repository,
+}
+
+// Initialise a git repo with a remote in the state that
+// remote - 3 commits, initial, first commit, second commit
+// local - 2 commits, initial, first commit
+// So that if we fetch on local then it should know about the second file
+pub fn initialise_git_repo(path: Option<&PathBuf>) -> TestGit {
+    let tmp_dir = TempDir::new("repo").expect("Temp dir created").into_path();
```

Comment:
`into_path` means the directory never gets cleaned up, even after the program exists (unlike in R I think).
I would store the `TempDir` object inside the `TestGit` struct, then when that struct goes out of scope the folder is removed.